src/providers/twitch/api/Helix.cpp

```diff
@@ -315,6 +315,46 @@ void Helix::getGameById(QString gameId,
         failureCallback);
 }
 
+void Helix::createClip(QString channelId,
+                       ResultCallback<HelixClip> successCallback,
+                       HelixFailureCallback failureCallback)
+{
+    QUrlQuery urlQuery;
+    urlQuery.addQueryItem("broadcaster_id", channelId);
+
+    this->makeRequest("clips", urlQuery)
+        .type(NetworkRequestType::Post)
+        .header("Content-Type", "application/json")
+        .onSuccess([successCallback, failureCallback](auto result) -> Outcome {
+            auto root = result.parseJson();
+            auto data = root.value("data");
+
+            if (!data.isArray())
+            {
+                failureCallback();
+                return Failure;
+            }
+
+            std::vector<HelixClip> clips;
```

Comment:
I'm wondering why this is a vector. Is there a possibility for the endpoint to return more than one clip? (I don't see any mention of this at https://dev.twitch.tv/docs/api/reference#create-clip)

Reply:
At least from what docs say, there's no possibility for this endpoint to return more than 1 clip, however due to lack of professional knowledge on how to handle the array I just followed an example from `Helix::fetchUsers`. I Will look into just getting the first element of `data` object from response and return it.